ReactiveCocoa/CocoaAction.swift

```diff
@@ -37,9 +35,11 @@ public final class CocoaAction<Sender>: NSObject {
 			producer.start()
 		}
 
-		isEnabled = action.isEnabled
-		isExecuting = action.isExecuting
-		
+		isEnabled = Property(initial: action.isEnabled.value,
+		                     then: action.isEnabled.producer.observe(on: UIScheduler()))
```

Comment:
It is safe here, as `property.producer` would always have the latest value overwritten the initial value should any race happen anyway. `PropertyProtocol.observe(on:)` makes sense too.